language/tools/move-package/src/source_package/manifest_parser.rs

```diff
@@ -306,155 +305,118 @@ fn parse_address_literal(address_str: &str) -> Result<AccountAddress, AccountAdd
-                        git_url: Symbol::from(git_url),
-                        git_rev: rev_name,
-                        subdir: subdir.clone(),
-                        download_to: local_path.clone(),
-                    });
-
-                    Ok(PM::Dependency {
-                        subst,
-                        version,
-                        digest,
-                        local: local_path.join(subdir),
-                        git_info,
-                        node_info,
-                    })
-                }
-                (None, None, Some(custom_key)) => {
-                    let package_name = Symbol::from(dep_name);
-                    let address = match table.remove("address") {
-                        None => bail!("Address not supplied for 'node' dependency"),
-                        Some(r) => Symbol::from(
-                            r.as_str()
-                                .ok_or_else(|| format_err!("Node address not a string"))?,
-                        ),
-                    };
-                    // Downloaded packages are of the form <sanitized_node_url>_<address>_<package>
-                    let node_url = custom_key
-                        .as_str()
-                        .ok_or_else(|| anyhow::anyhow!("Git URL not a string"))?;
-                    let local_path = PathBuf::from(MOVE_HOME.clone()).join(format!(
-                        "{}_{}_{}",
-                        url_to_file_name(node_url),
-                        address,
-                        package_name
-                    ));
-                    node_info = Some(PM::CustomDepInfo {
-                        node_url: Symbol::from(node_url),
-                        package_address: address,
-                        package_name,
-                        download_to: local_path.clone(),
-                    });
-                    Ok(PM::Dependency {
-                        subst,
-                        version,
-                        digest,
-                        local: local_path,
-                        git_info,
-                        node_info,
-                    })
-                }
-                _ => {
-                    let mut keys = vec!["local", "git"];
-                    if let Some(k) = custom_key_opt {
-                        keys.push(k.as_str())
-                    }
-                    let keys = keys
-                        .into_iter()
-                        .map(|s| format!("'{}'", s))
-                        .collect::<Vec<_>>();
-                    bail!(
-                        "must provide exactly one of {} for dependency.",
-                        keys.join(" or ")
-                    )
-                }
+fn parse_dependency(dep_name: &str, mut tval: TV) -> Result<PM::Dependency> {
+    let Some(table) = tval.as_table_mut() else {
+        bail!("Malformed dependency {}", tval);
+    };
+
+    let mut known_fields = vec![
```

Comment:
not needed now, but just wondering if this should a const that gets cloned

Reply:
Difficult to say whether this will make a big difference, given these are static strings, it's probably a wash.

Reply:
For me the const was more about intention and readability than perf, though I don't think it matters much

Reply:
Ah got it -- not super clear to me in this case, (I'm thinking of a scenario where I see the `const` and assume that the set of known fields is static when in fact it can be updated in the presence of package hooks).